### PR TITLE
Add AdminRoute component

### PR DESCRIPTION
**Pull request by willwull**
This PR contains the new AdminRoute component, which is used for `/create-user`.

I also had to add flow type files for react-redux and react-router-dom.

Note: the reason I removed `node_modules` from the flow ignore was because Immutable.js has its flow types in node_modules.

### CODE REVIEW

File: app/containers/AdminRoute/index.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import * as React from 'react'
+import { connect } from 'react-redux'
+import { Route, Redirect } from 'react-router-dom'
+import { List, Map } from 'immutable'
+
+type State = {|
+  isAdmin: boolean,
+  loggedIn: boolean,
+  hasFetchedLoggedInUser: boolean,
+|}
```
Comment (mictab) [inline app/containers/AdminRoute/index.js]: Newline?

File: flow-typed/npm/react-redux_v5.x.x.js [hunk 2]
```diff
@@ -0,0 +1,261 @@
+// flow-typed signature: 502cfd4f5e95c6308f747cdf16dc93ce
+// flow-typed version: 1751d5bf0a/react-redux_v5.x.x/flow_>=v0.68.0
+
+declare module "react-redux" {
```
Comment (mictab) [inline flow-typed/npm/react-redux_v5.x.x.js]: Do these need to be committed?

File: flow-typed/npm/react-router-dom_v4.x.x.js [hunk 3]
```diff
@@ -0,0 +1,180 @@
+// flow-typed signature: 53be1849af6037db65e90a7abc558afe
+// flow-typed version: f4e99ca1ed/react-router-dom_v4.x.x/flow_>=v0.63.x
+
+declare module "react-router-dom" {
```
Comment (mictab) [inline flow-typed/npm/react-router-dom_v4.x.x.js]: Do these need to be committed?

File: app/containers/AdminRoute/index.js (hunk 1 above)
Comment (willwull) [reply app/containers/AdminRoute/index.js]: Good catch 👍

File: flow-typed/npm/react-redux_v5.x.x.js (hunk 2 above)
Comment (willwull) [reply flow-typed/npm/react-redux_v5.x.x.js]: Yes, it seems so:

image
(https://flow.org/blog/2016/10/13/Flow-Typed/)

Comment (willwull) [reply flow-typed/npm/react-redux_v5.x.x.js]: Or at least, it's recommended, but I suppose we can choose not to.

File: flow-typed/npm/react-router-dom_v4.x.x.js (hunk 3 above)
Comment (willwull) [reply flow-typed/npm/react-router-dom_v4.x.x.js]: See above

File: flow-typed/npm/react-redux_v5.x.x.js (hunk 2 above)
Comment (mictab) [reply flow-typed/npm/react-redux_v5.x.x.js]: Cool

**Pull request merged**